# Helicobacter pylori induces short term trained immunity in monocytes via accumulation of NF-кB proteins
This repository complements the publication by Frauenlob et al (2023)

## Folders
- `data` : raw input data tables
- `analysis` : output tables generated by the scripts in this repository
- `scripts` : scripts for the analysis
- `figures`: generated plots
- `database`: downloaded gene set database used in fGSEA analysis

## Raw mass spectrometry data

The mass spectrometry proteomics data have been deposited to the ProteomeXchange Consortium  (http://www.proteomexchange.org/) via the PRIDE partner repository with the dataset identifier PXD045072

Prior to the analysis in this repository, the mass spectrometry proteomics data was treated in the following way:

For data evaluation, MaxQuant 2.0.1.0 (22) was used with the setting Reporter ion MS2, using weighted ratio to reference channel for normalization and correcting for isotope impurities in the TMTpro reagents. For protein identification, a database from the Uniprot consortium (23) including only reviewed Swiss-Prot entries for Homo sapiens (Human) from 10.03.2023 was used applying a 1 % false discovery rate. The obtained protein groups (MaxQuant output file “proteinGroups.txt”) were further processed using the Perseus 1.6.14.0 software platform (24): First the reporter intensities were log2 transformed, followed by the removal of proteins that were not detected in all samples, as well as the removal of proteins that were only identified by site, reverse sequence matches and potential contaminants like various isoforms of keratin. Next the reporter ion intensities were normalized by subtraction of the median intensity in each sample followed by a principal component analysis (PCA). As the PCA indicated a clustering not only dependent on the treatment, but also on the donor of the monocytes, the intensities were batch corrected employing the Remove batch effect function with the method Limma (25) embedded in the PerseusR plugin (26). 

Such log2-transformed, normalized, and batch-corrected data tables can be downloaded from the Zenodo repository xx.

## Analysis description

This repository contains scripts for the following analysis:

Differential expression analysis was performed with Limma employing R (version 4.2.2) in RStudio (version 2023.03.1+446), and p-value adjustment was conducted with the Benjamini-Hochberg procedure. Fast gene set enrichment analysis was performed using fgsea (27) package in R (version 4.3.0). The following databases were downloaded for the fgsea analysis: WikiPathways 2019 Human, NCI-Nature 2016, TRRUST Transcription Factors 2019, MSigDB Hallmark 2020, GO Cellular Component 2018, CORUM, KEGG 2019 Human, TRANSFAC and JASPAR PWMs, ENCODE and ChEA Consensus TFs from ChIP-X, GO Biological Process 2018, GO Molecular Function 2018, Human Gene Atlas. 


## Main workflow

From the Zenodo repository xx, download the intermediary data to folder `data`. Run these R scripts in the given in order to generate all files & plots

-   [analyse_dge.R](analyse_dge.R) - Loads log2 transformed, normalized, & batch-corrected dataset and analyses differential expression of proteins in treated (LPS, Hp, Aci) and uninduced monocytes
-   [monocytes_exploration_dge_heatmaps.R](monocytes_exploration_dge_heatmaps.R) - Loads log2 transformed, normalized, & batch-corrected dataset and perfoms explorative data analysis (PCA, Pearson correlations), loads results of DGE analysis and plots heatmaps of raw expressions only for significantly regulated proteins in lps_vs_uninduced, hp_vs_uninduced, aci_vs_uninduced conditions
-   [download_enrichr_databases.R](download_enrichr_databases.R) - Downloads respective databases from Enrichr for fGSEA analysis
-   [monocytes_gsea.R](monocytes_gsea.R) - Runs fGSEA analysis on all data 
-   [plot_gsea.R](monocytes_gsea.R) - Plots NES of fGSEA analysis 
